### PR TITLE
feat(toast): Change toast styles to latest iOS 14 styles

### PR DESCRIPTION
**Pull request by BerkeAras**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [X] Build (`npm run build`) was run locally and any changes were pushed
- [X] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: #22148

## What is the new behavior?

- The border-radius of toasts on iOS are now rounded (like iOS 14 style).
- The new property 'centeredText' has been added (iOS 14 style too).

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: core/src/components.d.ts [hunk 1]
```diff
@@ -2535,6 +2535,10 @@ export namespace Components {
           * An array of buttons for the toast.
          */
         "buttons"?: (ToastButton | string)[];
+        /**
+          * If `true`, the text of the toast will be centered.
+         */
+        "centeredText": boolean;
```
Comment (liamdebeasi) [inline core/src/components.d.ts:2541]: This prop should not be needed. If we just set `text-align: center` on `:host` in `toast.ios.scss` that should be enough. Users can override it if they want the text aligned another way.

**Review by liamdebeasi — CHANGES_REQUESTED** [review]
Thanks for the PR. Does this also make the toast appear at the top by default?

Just as a heads up: We are adding iOS 14 style updates in the next major release of Ionic Framework, so this will not get merged until then.

File: core/src/components.d.ts (hunk 1 above)
Comment (liamdebeasi) [reply core/src/components.d.ts:2541]: Actually not sure how this affects buttons, but either way we shouldn't need a prop to do this.

Comment (BerkeAras) [reply core/src/components.d.ts:2541]: @liamdebeasi Buttons are centered too.

Comment (BerkeAras) [reply core/src/components.d.ts]: @liamdebeasi See commit dc709a5.
The property **centeredText** has been removed. Text will be centered automatically and the font will be bold on iOS.

File: core/api.txt [hunk 2]
```diff
@@ -1259,7 +1259,6 @@ ion-title,css-prop,--color
 ion-toast,shadow
 ion-toast,prop,animated,boolean,true,false,false
 ion-toast,prop,buttons,(string | ToastButton)[] | undefined,undefined,false,false
-ion-toast,prop,centeredText,boolean,false,false,false
```
Comment (BerkeAras) [inline core/api.txt]: @liamdebeasi Property has been removed

File: core/src/components.d.ts (hunk 1 above)
Comment (liamdebeasi) [reply core/src/components.d.ts]: > @liamdebeasi Buttons are centered too.

Do you know if they are centered on native iOS 14 apps?

File: core/src/components/toast/toast.ios.scss [hunk 3]
```diff
@@ -22,6 +22,10 @@
   display: block;
   position: absolute;
 
+  font-weight: 600;
```
Comment (liamdebeasi) [inline core/src/components/toast/toast.ios.scss:25]: Can we move this to a sass variable in `toast.ios.vars.scss`?

**Review by BerkeAras — COMMENTED** [review]
@liamdebeasi Yes, toasts are centered on native iOS 14 too:
image

File: core/src/components/toast/toast.ios.scss (hunk 3 above)
Comment (BerkeAras) [reply core/src/components/toast/toast.ios.scss:25]: @liamdebeasi What about chaning the font-weight of the header to 600 too?
It does not look good if the header is thinner than everything else.

**Comment by appct** [comment]
Excellent. Thank you.

**Comment by liamdebeasi** [comment]
Thanks for the issue! We appreciate the work you put into creating this PR. After discussion with the team, we have decided not to merge the PR.

We did some investigation and found that this notification on iOS is used in privacy-focused system notifications. In other words, iOS sends this notification, not the app you are running. For example, this notification will appear when an app reads from your clipboard, camera roll, etc. It also appears when peripherals such as AirPods are connected.

We feel that updating our `ion-toast` style to mimic this system notification would lead to a confusing user experience and also risk apps getting rejected from the iOS App Store given the privacy-focused nature of the native notifications.

I am going to close this PR and the associated issue, but please let me know if you have any questions. Thank you!

**Pull request closed without merging**